message_app/user_sign_up.py

```diff
@@ -0,0 +1,48 @@
+from flask import Blueprint, request, session, redirect, url_for
+from message_app.utils import hashing
+from message_app.model import User
+import re
+
+user_sign_up = Blueprint("user_sign_up", __name__)
+
+@user_sign_up.route("/api/signup", methods=["POST"])
+def api_user_signup():
+    try:
+        params = request.form
+        username = params.get("username", "")
+        password = params.get("password", "")
+        confirmPassword = params.get("confirmPassword", "")
+        email = params.get("email", "")
+        emailRegex = r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
+
+        # Verify email
+        if re.fullmatch(emailRegex, email) is None:
```

Comment:
Should we leave the following on the front-end?
- email verification
- valid password (empty, non-matching when signing up)
This way, we can probably safe one request to the server. Just my thought. What do you think?

Reply:
Yes, we can leave the validation to the front-end. Flask has an extension called _**WTForm**_ which is very helpful for form validation.

Source:
https://flask.palletsprojects.com/en/2.0.x/patterns/wtforms/

Reply:
I believe the form validation could be fully implemented in JavaScript, so that we do not need to verify basic validity on the API side and install a new extension.

@bhy2001 @Hudson-Pufferfish Please check out this regex and implement it on the frontend.